src/shogun/mathematics/linalg/LinalgBackendEigen.h

```diff
@@ -284,6 +309,58 @@ class LinalgBackendEigen : public LinalgBackendBase
 		result_eig = alpha * a_eig + beta * b_eig;
 	}
 
+	/** Eigen3 Cholesky decomposition */
+	template <typename T>
+	SGMatrix<T> cholesky_factor_impl(const SGMatrix<T>& a, const bool lower) const
+	{
+		SGMatrix<T> c(a.num_rows, a.num_cols);
+		set_const_impl<T>(c, 0);
+		typename SGMatrix<T>::EigenMatrixXtMap a_eig = a;
+		typename SGMatrix<T>::EigenMatrixXtMap c_eig = c;
+
+		Eigen::LLT<Eigen::Matrix<T,Eigen::Dynamic,Eigen::Dynamic> > llt(a_eig);
+
+		//compute matrix L or U
+		if(lower==false)
+			c_eig = llt.matrixU();
+		else
+			c_eig = llt.matrixL();
+
+		/*
+		 * checking for success
+		 *
+		 * 0: Eigen::Success. Decomposition was successful
+		 * 1: Eigen::NumericalIssue. The provided data did not satisfy the prerequisites.
+		 */
+		REQUIRE(llt.info()!=Eigen::NumericalIssue, "Matrix is not Hermitian positive definite!\n");
+
+		return c;
+	}
+
+	/** Eigen3 Cholesky solver */
+	template <typename T>
+	SGVector<T> cholesky_solver_impl(const SGMatrix<T>& a, const SGVector<T>& b) const
```

Comment:
@karlnapf Do you mean that users explicitly call `cholesky_factor` and pass the matrix to `cholesky_solver`? I saw that's how scipy achieved `cholesky_solver`. But they called `lapack::potrs` directly in the function, which can solve `Lu=b`, while in `Eigen3`, the solver is wrapped in `cholesky_factor`. Here I just created a new `llt` instance and it came with the triangular solver.  I think we will have to code back substitution from scratch if we don't call `eigen3.llt.solve()`.